numpy/polynomial/polyutils.py

```diff
@@ -118,25 +118,28 @@ def as_series(alist, trim=True):
     for a in arrays:
         if a.size == 0:
             raise ValueError("Coefficient array is empty")
-    if any(a.ndim != 1 for a in arrays):
-        raise ValueError("Coefficient array is not 1-d")
+        if a.ndim != 1:
+            raise ValueError("Coefficient array is not 1-d")
     if trim:
         arrays = [trimseq(a) for a in arrays]
 
-    if any(a.dtype == np.dtype(object) for a in arrays):
+    try:
+        dtype = np.common_type(*arrays)
```

Comment:
Why can't we just use `result_type` and be done with it!?  Does this actually need any of the `common_type` shenanigans that don't really make sense?

Reply:
Maybe we can. There was another PR where I tried to replace `common_type` with `result_type`. It looked promising, but in the end it didn't work out. Maybe here it does work, I'll check at some point